packages/jasmine-parameterized/src/parameterize.ts

```diff
@@ -0,0 +1,83 @@
+import { Spec, SpecOverride, Suite, SuiteOverride } from './types.js';
+
+/**
+ * Used to create a parameterized test or suite using an object of names and arbitrary test values.
+ * In the following example:
+ *  - the test named `catsAndDogs` is focused for debugging
+ *  - the test named `frogs` is configured to always be disabled
+ *  - the test named `men` will run normally as it has no override
+ * @example
+ * const rainTests = {
+ *     catsAndDogs: 'idiom',
+ *     frogs: 'idiom',
+ *     men: 'lyrics'
+ * } as const;
+ * describe('Different rains', () => {
+ *     parameterize(rainTests, (spec, name, value) => {
+ *         spec(`of type ${name} exist`, () => {
+ *             expect(value).toBeDefined();
+ *         });
+ *     }, it, {
+ *         catsAndDogs: fit,
+ *         frogs: xit
+ *     });
+ * });
+ */
+export function parameterize<T extends object>(
+    testCases: T,
+    test: (spec: Spec, name: keyof T, value: T[keyof T]) => void,
+    defaultTestType: Spec,
+    overrides?: {
+        [P in keyof T]?: SpecOverride;
+    }
+): void;
+export function parameterize<T extends object>(
+    testCases: T,
+    test: (spec: Suite, name: keyof T, value: T[keyof T]) => void,
+    defaultTestType: Suite,
+    overrides?: {
+        [P in keyof T]?: SuiteOverride;
+    }
+): void;
+export function parameterize<T extends object, U = Spec | Suite>(
```

Comment:
Two nits/pref

- Can we make the first parameter the type selector? i.e. make defaultTestType first? Then it'll feel more like the parameterize specific options are first followed by everything else it delegates to instead of being in the middle
- Instead of defaultTestType being a reference to either it or describe could we use an enum or even just a union of two string options? `'spec' | 'suite'`. I think an enum / string union is easier to do an exhaustiveness check over while lots of different types fulfill the contract of `defaultTestType: U`